### ISSUE TITLE
User Info loading to blank page

### ISSUE THREAD

**Issue by UAnton** [RocketChat/Rocket.Chat#18398]
### Description:

Can't change user settings. User preference not showing.

### Steps to reproduce:

1. Go to Administration - Users
2. Click on user

### Expected behavior:

Blank page. 

### Server Setup Information:

- Version of Rocket.Chat Server: 3.5.0
- Operating System: Ubuntu 20.04
- Deployment Method: manual
- Number of Running Instances: 1
- DB Replicaset Oplog: 
- NodeJS Version: 12.14.0
- MongoDB Version: 4.0.19

### Client Setup Information

- Desktop App or Browser Version: 2.17.9
- Operating System: MacOS

### Relevant logs:
From browser console:
`TypeError: undefined is not an object (evaluating 'g.user.roles.includes')`

**Comment by MartinSchoeler** [comment]
I'm unable to reproduce this issue, what roles do your user have? Any custom roles? Thanks!

**Comment by phlbrz** [comment]
Hi,

Same or similar problem here.

https://myexamplechat.com/admin/users/edit/xxxxXXxx

I noticed another similar problem at:

https://myexamplechat.com/account/tokens
https://myexamplechat.com/account/profile
https://myexamplechat.com/account/preferences

For this second problem, showed some errors in chrome console:

```
98cbd60f7c9f791e528b7293c7741c4255338a15.js?meteor_js_resource=true:473 TypeError: Cannot convert undefined or null to object
    at Function.values (<anonymous>)
    at eval (/client/components/CustomFieldsForm.js:1)
    at Object.useMemo (98cbd60f7c9f791e528b7293c7741c4255338a15.js?meteor_js_resource=true:473)
    at n.useMemo (98cbd60f7c9f791e528b7293c7741c4255338a15.js?meteor_js_resource=true:449)
    at b (/client/components/CustomFieldsForm.js:1)
    at po (98cbd60f7c9f791e528b7293c7741c4255338a15.js?meteor_js_resource=true:473)
    at Zo (98cbd60f7c9f791e528b7293c7741c4255338a15.js?meteor_js_resource=true:473)
    at us (98cbd60f7c9f791e528b7293c7741c4255338a15.js?meteor_js_resource=true:473)
    at ds (98cbd60f7c9f791e528b7293c7741c4255338a15.js?meteor_js_resource=true:473)
    at Xs (98cbd60f7c9f791e528b7293c7741c4255338a15.js?meteor_js_resource=true:473)

```
Some line from chrome `/client/components/CustomFieldsForm.js:formatted`

```
          , i = u(()=>Object.values(n).map(e=>[e, e]), [n])
```

**Comment by BizClimb** [comment]
I can confirm I'm having the same issue after upgrading to 3.5, I've noticed it occurs only if I click a default user that registered but hasn't been edited yet (that has the default **user** role) the admin area goes white. If you are in the browser, you can REFRESH the browser and it loads the user card but all data is missing (username, etc...) but then you can click the EDIT button and it lets you in to edit the user.

This does not happen on users with already modified roles (I modify all users from the default user role to another role).

Here is the console error when it goes white:
98cbd60f7c9f791e528b7293c7741c4255338a15.js?meteor_js_resource=true:473 Invariant Violation: Minified React error #31; visit https://reactjs.org/docs/error-decoder.html?invariant=31&args[]=object%20with%20keys%20%7Blabel%2C%20value%7D&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings. 
    at i (https://chat.mysite.com/98cbd60f7c9f791e528b7293c7741c4255338a15.js?meteor_js_resource=true:473:296)
    at s (https://chat.mysite.com/98cbd60f7c9f791e528b7293c7741c4255338a15.js?meteor_js_resource=true:473:566)
    at Ra (https://chat.mysite.com/98cbd60f7c9f791e528b7293c7741c4255338a15.js?meteor_js_resource=true:473:50079)
    at https://chat.mysite.com/98cbd60f7c9f791e528b7293c7741c4255338a15.js?meteor_js_resource=true:473:54856
    at Bo (https://chat.mysite.com/98cbd60f7c9f791e528b7293c7741c4255338a15.js?meteor_js_resource=true:473:61787)
    at Zo (https://chat.mysite.com/98cbd60f7c9f791e528b7293c7741c4255338a15.js?meteor_js_resource=true:473:69721)
    at us (https://chat.mysite.com/98cbd60f7c9f791e528b7293c7741c4255338a15.js?meteor_js_resource=true:473:91128)
    at ds (https://chat.mysite.com/98cbd60f7c9f791e528b7293c7741c4255338a15.js?meteor_js_resource=true:473:91515)
    at Xs (https://chat.mysite.com/98cbd60f7c9f791e528b7293c7741c4255338a15.js?meteor_js_resource=true:473:98599)
    at Qs (https://chat.mysite.com/98cbd60f7c9f791e528b7293c7741c4255338a15.js?meteor_js_resource=true:473:97979)

**Comment by MartinSchoeler** [comment]
@UAnton @BizClimb @phlbrz Do you have any custom fields set up?

**Comment by UAnton** [comment]
Hi Martin, 
No, this clean basic installation. One admin user and LDAP auth. No any changes.

**Comment by BizClimb** [comment]
> @UAnton @BizClimb @phlbrz Do you have any custom fields set up?

I do have 2 custom fields setup. There was similar behavior when the JSON syntax wasn't indented with TABS properly in previous versions, I fixed this by adding TABS (expected spacing) to the JSON code and saving and then it worked again, but I re-confirmed the custom fields were spaced properly, and I also even added a new custom field to see if it would somehow refresh and work properly, but still nothing.

So yes, I DO have custom fields setup.

**Comment by phlbrz** [comment]
@MartinSchoeler 

Yes, I do.

```
{   
    "title": "title",
    "telephoneNumber": "telephoneNumber",
    "division": "division",
    "extensionAttribute1": "extensionAttribute1",
    "extensionAttribute3": "extensionAttribute3",
    "streetAddress": "streetAddress",
    "manager": "manager"
}
```

Everything was working before 3.5.0.
First I always upgrade rocketchat in stage environment, test principal things and upgrade production, but this bug get me stuck until it solves first.
Before upgrade to 3.5.0 in stage, rocketchat version was 3.3.3 and worked.

I have a production instance in version 3.2.2 with this custom fields and it is working like a charm.

Thank's.

**Comment by BizClimb** [comment]
I can also confirm that I've upgraded through 3-4 releases as well. It was working fine for me in 3.4.2 as well as the previous versions. I should note that as a test I spun up a new rocket chat release 3.5, set the custom fields up, and it's working fine. So this leads me to believe something from at least 2 releases ago has caused the issue to become prevalent now in 3.5. This is unconfirmed, just speculation based on the information being gathered. Let me know if you need any more info from me or anything to test. Thanks.

**Comment by MartinSchoeler** [comment]
I see, Yeah we did some heavy refactoring on the user info screens, It could be an issue with previous set custom fields. I'll try to test out more to see what we find out. Probably will need a migration for this. I'll keep you all updated about it

**Comment by Endareth** [comment]
Can confirm seeing this issue building from current develop branch (55703f191f4a5a1230ce0fea3c536fee6d769277) on a clean database. Issue appears to be with the parameters on the `users.list` API. Passing no parameters is a successful call, but as soon as any parameters are added (such as the UI does), a timeout occurs. Of note is that the logger call at app/api/server/api.js:348 doesn't even run, so I suspect something about that function creation block is resulting in a badly defined function or something (not a js coder, so not 100% on this).

**Comment by phlbrz** [comment]
@MartinSchoeler 

Some workaround for this line:
  
https://github.com/RocketChat/Rocket.Chat/blob/1bed2977cac7416741380ebfd4b41ce611d47545/client/components/CustomFieldsForm.js#L29

Could be:

```
options =  (options) ? options : "";
const mappedOptions = useMemo(() => Object.values(options).map((value) => [value, value]), [options]);
```
I don't know where options var come from the code, so just a guess.

Tested in https://playcode.io/ and `TypeError: Cannot convert undefined or null to object` solved.

PS.: I'm not a javascript developer.

:smile:

**Comment by UAnton** [comment]
3.5.1-  problem still here. This happened after enabling LDAP sync. Before enable - all was ok.

**Comment by UAnton** [comment]
Any news?

**Comment by UAnton** [comment]
3.5.2 - problem still here. This happened after enabling LDAP auth and sync. Before enable - all was ok.

```
e49465247681928f62c29836b6a5fe7e1714f082.js?meteor_js_resource=true:473 TypeError: Cannot read property 'includes' of undefined
    at x (/client/admin/users/UserInfo.js:1)
    at po (e49465247681928f62c29836b6a5fe7e1714f082.js?meteor_js_resource=true:473)
    at Ho (e49465247681928f62c29836b6a5fe7e1714f082.js?meteor_js_resource=true:473)
    at Zo (e49465247681928f62c29836b6a5fe7e1714f082.js?meteor_js_resource=true:473)
    at us (e49465247681928f62c29836b6a5fe7e1714f082.js?meteor_js_resource=true:473)
    at ds (e49465247681928f62c29836b6a5fe7e1714f082.js?meteor_js_resource=true:473)
    at Xs (e49465247681928f62c29836b6a5fe7e1714f082.js?meteor_js_resource=true:473)
    at Qs (e49465247681928f62c29836b6a5fe7e1714f082.js?meteor_js_resource=true:473)
    at Hs (e49465247681928f62c29836b6a5fe7e1714f082.js?meteor_js_resource=true:473)
    at hs (e49465247681928f62c29836b6a5fe7e1714f082.js?meteor_js_resource=true:473)
```

**Comment by BizClimb** [comment]
This is still happening for me as well in 3.5.2. I do not have LDAP enabled, but I do have custom registration fields.

**Comment by ZeroPoke** [comment]
I noticed this on my RocketChat install via snap I did today.  That also does the blank user info.

But it only does that for users with a role which happens to be all my users imported form LDAP.
Altho users that do have a role the user info will load for them.

**Comment by BizClimb** [comment]
At first this started only for users on the default role, and then users I moved over to a new role manually were ok. But today I noticed it happening on some users who had changed roles already.

**Comment by UAnton** [comment]
I tried to install older versions but have the same problem and many LDAP errors like:
`I20200826-20:00:00.414(4) server.js:204 LDAPSync ➔ error Error creating user errorClass [Error]: Username already exists. [403]     at handleError (packages/accounts-password/password_server.js:110:17)     at checkForCaseInsensitiveDuplicates (packages/accounts-password/password_server.js:257:7)     at createUser (packages/accounts-password/password_server.js:1116:3)     at AccountsServer.Accounts.createUser (packages/accounts-password/password_server.js:1188:10)     at addLdapUser (app/ldap/server/sync.js:463:29)     at app/ldap/server/sync.js:529:5     at Array.forEach (<anonymous>)     at app/ldap/server/sync.js:496:13     at runWithEnvironment (packages/meteor.js:1286:24) {   isClientSafe: true,   error: 403,   reason: 'Username already exists.',   details: undefined,   message: 'Username already exists. [403]',   errorType: 'Meteor.Error' }`

**Comment by BizClimb** [comment]
This issue has been resolved for me since 3.5.3

**Comment by UAnton** [comment]
> This issue has been resolved for me since 3.5.3

Today I tried to install versions from 3.4.2 to 3.5.4 - no changes :(

**Comment by mosesdd** [comment]
This is still happening for me in 3.5.2 and 3.6.0 as well

**Comment by mosesdd** [comment]
same for 3.6.1

**Comment by mosesdd** [comment]
same for 3.6.2

**Comment by willoughby** [comment]
also have this in 3.70 is getting very annoying for my customers, I really dont want to go back to Discord.

**Comment by pierre-lehnen-rc** [comment]
Should've been fixed on 3.61. by #18851. 
If you're still having this problem, please share your error logs.

**Comment by phlbrz** [comment]
@pierre-lehnen-rc Solved for me!

Everything working as expected!

Thank's!

**Comment by phlbrz** [comment]
Hi,

After update to 3.7.0 my server cannot show my avatars.
Everytime I open /admin/users, avatars shows as code 403 in chrome.
Rocketchat logs `The server detected an unauthenticated access to an user avatar. This type of request will soon be blocked by default.`
If I'm admin and logged in, I don't understand why it's happening.
Already opened an incognito mode, same thing.

![Screenshot from 2020-10-01 18-43-46](https://user-images.githubusercontent.com/6169365/94866650-21ee7780-0416-11eb-9292-5639df9ab097.png)

![Screenshot from 2020-10-01 18-43-34](https://user-images.githubusercontent.com/6169365/94866652-22870e00-0416-11eb-9765-c978d768aa64.png)

**Comment by Endareth** [comment]
On 3.6.2, still experiencing this issue (or at least I thought it was the identical issue).
Going to Administration -> Users fails to load, with a red Undefined popup.
Console error:
```
GET https://im.REDACTED.com/api/v1/users.list?fields={%22name%22:1,%22username%22:1,%22emails%22:1,%22roles%22:1,%22status%22:1,%22avatarETag%22:1}&query={%22$or%22:[{%22emails.address%22:{%22$regex%22:%22%22,%22$options%22:%22i%22}},{%22username%22:{%22$regex%22:%22%22,%22$options%22:%22i%22}},{%22name%22:{%22$regex%22:%22%22,%22$options%22:%22i%22}}]}&sort={%22name%22:1,%22usernames%22:1}&count=25 502 (Proxy Error)
send @ 81bba22dadcf0237c74ba4bd697dc8d8666e87f5.js?meteor_js_resource=true:48
ajax @ 81bba22dadcf0237c74ba4bd697dc8d8666e87f5.js?meteor_js_resource=true:48
l @ 81bba22dadcf0237c74ba4bd697dc8d8666e87f5.js?meteor_js_resource=true:423
_jqueryCall @ 81bba22dadcf0237c74ba4bd697dc8d8666e87f5.js?meteor_js_resource=true:423
get @ 81bba22dadcf0237c74ba4bd697dc8d8666e87f5.js?meteor_js_resource=true:423
get @ 81bba22dadcf0237c74ba4bd697dc8d8666e87f5.js?meteor_js_resource=true:423
c @ /client/providers/ServerProvider.js:1
eval @ /client/contexts/ServerContext.ts:1
s @ /client/hooks/useEndpointData.js:1
eval @ /client/hooks/useEndpointData.js:1
Ii @ /node_modules/react-dom/cjs/react-dom.production.min.js:219
Vj @ /node_modules/react-dom/cjs/react-dom.production.min.js:265
exports.unstable_runWithPriority @ /node_modules/scheduler/cjs/scheduler.production.min.js:27
cg @ /node_modules/react-dom/cjs/react-dom.production.min.js:130
Dj @ /node_modules/react-dom/cjs/react-dom.production.min.js:265
eval @ /node_modules/react-dom/cjs/react-dom.production.min.js:264
X @ /node_modules/scheduler/cjs/scheduler.production.min.js:25
H.port1.onmessage @ /node_modules/scheduler/cjs/scheduler.production.min.js:22
/client/hooks/useEndpointData.js:1 Error: Proxy Error
    at originalCallback (81bba22dadcf0237c74ba4bd697dc8d8666e87f5.js?meteor_js_resource=true:423)
    at Object.processTwoFactorError (81bba22dadcf0237c74ba4bd697dc8d8666e87f5.js?meteor_js_resource=true:423)
    at Object.a [as error] (81bba22dadcf0237c74ba4bd697dc8d8666e87f5.js?meteor_js_resource=true:423)
    at c (81bba22dadcf0237c74ba4bd697dc8d8666e87f5.js?meteor_js_resource=true:48)
    at Object.fireWith [as rejectWith] (81bba22dadcf0237c74ba4bd697dc8d8666e87f5.js?meteor_js_resource=true:48)
    at w (81bba22dadcf0237c74ba4bd697dc8d8666e87f5.js?meteor_js_resource=true:48)
    at XMLHttpRequest.<anonymous> (81bba22dadcf0237c74ba4bd697dc8d8666e87f5.js?meteor_js_resource=true:48)
```

**Comment by qxpat** [comment]
I think this might be the same issue (no email address for users): https://github.com/RocketChat/Rocket.Chat/issues/18435#issuecomment-698482010

**Comment by MartinSchoeler** [comment]
> Hi,
> 
> After update to 3.7.0 my server cannot show my avatars.
> Everytime I open /admin/users, avatars shows as code 403 in chrome.
> Rocketchat logs `The server detected an unauthenticated access to an user avatar. This type of request will soon be blocked by default.`
> If I'm admin and logged in, I don't understand why it's happening.
> Already opened an incognito mode, same thing.
> 
> It happens in https://myinstance.chat.xxx/directory/users also.
> 
> I disabled Accounts -> Block Unauthenticated Access to Avatars, but still happening. I don't know if it has an issue already.
> 
> ![Screenshot from 2020-10-01 18-43-46](https://user-images.githubusercontent.com/6169365/94866650-21ee7780-0416-11eb-9292-5639df9ab097.png)
> 
> ![Screenshot from 2020-10-01 18-43-34](https://user-images.githubusercontent.com/6169365/94866652-22870e00-0416-11eb-9765-c978d768aa64.png)
> 
> https://chat.xxx/avatar/my_id -> shows Forbidden
> https://chat.xxx/avatar/my_id?etag=xxxxx -> shows avatars
> 
> My Roles:
> 
> ![Screenshot from 2020-10-02 14-15-12](https://user-images.githubusercontent.com/6169365/94950970-d5f50e80-04b9-11eb-9f1e-b75bb37d755d.png)
> 
> * user
> * Admin
> * Livechat Agent
> * Livechat Manager
> 
> I looked for some issue about that, I don't know if it's better to open a new one for 3.7.0, but it happened after update from
> 3.5.0.
> 
> Channel also, same problem, never show user avatar.

Hello can you open a new issue for this problem, might not be related at all, also be sure to upgrade to 3.8.0 when is out, we did some improvements on the avatar providers. Thanks!

**Comment by MartinSchoeler** [comment]
@Endareth Could you test with version 3.7.0+ Also do you get any errors on the server console? Thanks!

**Comment by phlbrz** [comment]
> > Hi,
> > After update to 3.7.0 my server cannot show my avatars.
> > Everytime I open /admin/users, avatars shows as code 403 in chrome.
> > Rocketchat logs `The server detected an unauthenticated access to an user avatar. This type of request will soon be blocked by default.`
> > If I'm admin and logged in, I don't understand why it's happening.
> > Already opened an incognito mode, same thing.
> > It happens in https://myinstance.chat.xxx/directory/users also.
> > I disabled Accounts -> Block Unauthenticated Access to Avatars, but still happening. I don't know if it has an issue already.
> > ![Screenshot from 2020-10-01 18-43-46](https://user-images.githubusercontent.com/6169365/94866650-21ee7780-0416-11eb-9292-5639df9ab097.png)
> > ![Screenshot from 2020-10-01 18-43-34](https://user-images.githubusercontent.com/6169365/94866652-22870e00-0416-11eb-9765-c978d768aa64.png)
> > https://chat.xxx/avatar/my_id -> shows Forbidden
> > https://chat.xxx/avatar/my_id?etag=xxxxx -> shows avatars
> > My Roles:
> > ![Screenshot from 2020-10-02 14-15-12](https://user-images.githubusercontent.com/6169365/94950970-d5f50e80-04b9-11eb-9f1e-b75bb37d755d.png)
> > 
> > * user
> > * Admin
> > * Livechat Agent
> > * Livechat Manager
> > 
> > I looked for some issue about that, I don't know if it's better to open a new one for 3.7.0, but it happened after update from
> > 3.5.0.
> > Channel also, same problem, never show user avatar.
> 
> Hello @phlbrz can you open a new issue for this problem, might not be related at all, also be sure to upgrade to 3.8.0 when is out, we did some improvements on the avatar providers. Thanks!

Hello @MartinSchoeler , @ggazzo reopened https://github.com/RocketChat/Rocket.Chat/issues/17819, I'm ok with the description, also linked both issues. Thank you!

**Comment by MartinSchoeler** [comment]
We did a ton of work to fix these blank pages on the user info, if any of you still face this issue with the latest develop version (or the upcoming release) please let me know. Thanks for all the feedback!

**Comment by mosesdd** [comment]
Still the same problem with 3.8.0

**Comment by Endareth** [comment]
> @Endareth Could you test with version 3.7.0+ Also do you get any errors on the server console? Thanks!

Finally able to test with latest snap (3.8.1), and still the same behaviour (undefined popup, page fails to load) for the Users scren. No log activity on the server at all. Still suspect root cause here is from an LDAP sync'd user account coming in with some dodgy data, but hard to check.

**Comment by mosesdd** [comment]
Still the same problem with 3.9.0

**Comment by Endareth** [comment]
Happy to say that issues seems to finally be resolved in 3.9.3! Thanks whoever managed to fix this, I was getting tired of using curl for all my user management (but at least I *could* do it, thanks for the API)!

**Comment by mosesdd** [comment]
I can not confirm this, still the same problem for me with 3.9.4

**Comment by mosesdd** [comment]
still the same with 3.10.5

**Comment by Basili0** [comment]
Same here with 3.11.0. Clean install, users.info fails with 403 for an admin user.

**Comment by MartinSchoeler** [comment]
@Basili0 @mosesdd do you see any errors on the browser or server console when the issue happens. Also could you give us more info on what is happening (any errors popup or anything that catches your eye). Thanks!

**Comment by jrgong420** [comment]
Same issue here with some users

**Comment by mosesdd** [comment]
@MartinSchoeler I can see a 404 in Browser Console when I try to open the user info page. Seems like trying to load an avatar of a user (that is disabled in ldap) fails
https://rocketchaturl/avatar/username

**Comment by mosesdd** [comment]
this seems to be fixed for me in 3.13.3

**Comment by FravashY** [comment]
Same issue here with self-hosted 3.14.0

**Comment by dougfabris** [comment]
Between version 3.14 and the actual, we did a bunch of fixes in User Info, and we can't reproduce anymore. I'm Closing this one! Thanks all for the feedback!

**Comment by dougfabris** [comment]
> Still the same problem with 4.1.0

@rtest12 can you provide more details? e.g screenshots

**Comment by rtest12** [comment]
> > Still the same problem with 4.1.0
> 
> @rtest12 can you provide more details? e.g screenshots

Sure
Apparently, this happens precisely when authorizing through a keycloak, if I go as a standard administrator, then everything opens. The screenshot probably makes no sense, just a gray background on the whole screen.

**Comment by rtest12** [comment]
```
.loading-animation {  
  background: url(https://example.com/images/preloader.gif)
  background-repeat: no-repeat;
  background-position: 50% 50%
}

#react-root .page-loading .loading-animation .bounce {
  display: none;
  height: 0px;
}
```

What am I doing wrong?
